synfig-studio/src/gui/dialogs/dialog_guide.cpp

```diff
@@ -0,0 +1,261 @@
+	rotationAngleLabel->set_halign(Gtk::ALIGN_CENTER);
+	rotationAngleLabel->set_mnemonic_widget(*angle_widget);
+	guideGrid->attach(*rotationAngleLabel, 0, 0, 1, 1);
+	guideGrid->attach(*angle_widget      , 1, 0, 1, 1);
+	guideGrid->attach(angle_type_picker  , 2, 0, 1, 1);
+
+/*	auto coordGrid = manage(new Gtk::Grid());
+	coordGrid->get_style_context()->add_class("dialog-secondary-content");
+	coordGrid->set_row_spacing(6);
+	coordGrid->set_column_spacing(8);
+
+	Gtk::Frame *coordFrame = manage(new Gtk::Frame(_("Rotate By Setting Coordinates")));
+	coordFrame->set_shadow_type(Gtk::SHADOW_NONE);
+	((Gtk::Label *) coordFrame->get_label_widget())->set_markup(_("<b>Rotate By Setting Coordinates</b>"));
+	coordFrame->set_margin_bottom(5);
+	coordFrame->set_margin_top(5);
+
+	Gtk::Label *centerCoordLabel = manage(new Gtk::Label(_("_Center Point"), true));
+	centerCoordLabel->set_halign(Gtk::ALIGN_CENTER);
+	centerCoordLabel->set_margin_end(15);
+	centerCoordLabel->set_mnemonic_widget(*center_x_widget);
+	coordGrid->attach(*centerCoordLabel, 0, 0, 1, 1);
+	coordGrid->attach(*center_x_widget   , 1, 0, 1, 1);
+	coordGrid->attach(*center_y_widget   , 2, 0, 1, 1);
+
+	Gtk::Label *otherCoordLabel = manage(new Gtk::Label(_("_Other Point"), true));
+	otherCoordLabel->set_mnemonic_widget(*point_x_widget);
+	otherCoordLabel->set_margin_end(15);
+	coordGrid->attach(*otherCoordLabel, 0, 1, 1, 1);
+	coordGrid->attach(*point_x_widget,  1, 1, 1, 1);
+	coordGrid->attach(*point_y_widget,  2, 1, 1, 1);
+*/
+
+	guide_box->add(*angleFrame);
+	guide_box->add(*guideGrid);
+//	guide_box->add(*coordFrame);
+//	guide_box->add(*coordGrid);
+	guide_box->set_margin_bottom(5);
+
+	//Box end
+	get_content_area()->pack_start(*guide_box);
+
+	Gtk::Button *apply_button(manage(new Gtk::Button(_("_Apply"), true)));
+	apply_button->show();
+	add_action_widget(*apply_button, Gtk::RESPONSE_APPLY);
+	apply_button->signal_clicked().connect(sigc::bind(sigc::mem_fun(*this, &Dialog_Guide::on_ok_or_apply_pressed), false));
+
+	Gtk::Button *ok_button(manage(new Gtk::Button(_("_OK"), true)));
+	ok_button->show();
+	add_action_widget(*ok_button, Gtk::RESPONSE_OK);
+	ok_button->signal_clicked().connect(sigc::bind(sigc::mem_fun(*this, &Dialog_Guide::on_ok_or_apply_pressed), true));
+
+	guide_box->show_all();
+
+//	current_work_area = work_area;
+}
+
+Dialog_Guide::~Dialog_Guide()
+{
+}
+
+void
+Dialog_Guide::set_current_guide(GuideList::iterator current_guide)
+{
+	curr_guide = current_guide;
+	init_widget_values();
+}
+
+
+void
+Dialog_Guide::on_ok_or_apply_pressed(bool ok)
+{
+	if (synfig::Angle::deg(curr_guide->angle).get() != angle_widget->get_value() && degrees) {
+		curr_guide->angle = synfig::Angle::deg(angle_widget->get_value());
+	} else if (curr_guide->angle.get() != angle_widget->get_value() && !degrees) {
+		curr_guide->angle = synfig::Angle::rad(angle_widget->get_value());
+	} /*else
+		set_new_coordinates();*/
```

Comment:
For readability (and not-needed efficiency), swap the condition order:
if (degrees && ...)